### ISSUE TITLE
"You can fix this by adding a `site.url` to your `nuxt.config` or a `NUXT_PUBLIC_SITE_URL`"

### ISSUE THREAD

**Issue by bgrand-ch** [harlan-zw/nuxt-simple-sitemap#136]
### Describe the bug

Hello,

Thanks for this Nuxt module!

I have the following error when the module version is higher than [v3.2.7](https://github.com/harlan-zw/nuxt-simple-sitemap/releases/tag/v3.2.7): 

![Screenshot 2023-09-15 at 08 16 22](https://github.com/harlan-zw/nuxt-simple-sitemap/assets/17108974/4f092d98-229a-4a4d-a148-4a99abcf854c)

My current Nuxt configuration:

![Screenshot 2023-09-15 at 08 32 06](https://github.com/harlan-zw/nuxt-simple-sitemap/assets/17108974/e157c1ec-7c19-48c9-aea0-697a214bc140)

![Screenshot 2023-09-15 at 08 31 29](https://github.com/harlan-zw/nuxt-simple-sitemap/assets/17108974/be75a1a1-8314-43a0-a6a4-a8363d82cb12)

I have all my env values when I log the `env` variable.

Can you help me solve this problem please?

### System / Nuxt Info

```Shell
- Operating System: Darwin
- Node Version:     v18.16.0
- Nuxt Version:     3.7.3
- CLI Version:      3.8.4
- Nitro Version:    2.6.3
- Package Manager:  pnpm@8.7.5
- Builder:          -
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
```

**Comment by harlan-zw** [comment]
Thanks for the detailed issue, not exactly sure what's going on here but I've made a few changes that may help:
- Fixed a small bug in how the config stack is resolved (no longer uses empty strings)
- Moved the assertion to when prerendering starts (in case the site config is getting set delayed for some reason)
- Disabled the `throwError` so it doesn't block you

Can you try out 3.3.3 and let me know how it goes, if the error is thrown and if the sitemap works. Thanks!

**Comment by bgrand-ch** [comment]
Thanks a lot for this fixes, build success with a sitemap on v3.3.3!
